test/rules/default.spec.ts

```diff
@@ -123,7 +130,7 @@ ruleTester.run('default', rule, {
       },
     }),
 
-    ...SYNTAX_CASES,
+    ...(SYNTAX_VALID_CASES as RunTests<typeof rule>['valid']),
```

Comment:
Is it possible to create a minimal type of `ValidTestCase` that is rule-agnostic? And we can type `SYNTAX_VALID_CASES` with that?

Reply:
It is already done on `utils`:
```
export const SYNTAX_VALID_CASES: TSESLintRunTests<string, unknown[]>['valid']
```
but this is will be never assignable to something that is rule specific like in these scenario where the `SYNTAX_VALID_CASES` is provided directly to a `RuleTester` instance. 
Maybe I can create a `createSyntaxValidCases` function that infers the type and avoid to performs an explicit casting but IMHO casting is fine:
we have a generic value that is assigned to something more specific so it can be safely casted to that specific value.

Reply:
Then it would be fine~